apps/app/src/pages/Build/index.js

```diff
@@ -128,6 +148,30 @@ function BuildStat({ status, count, label }) {
   );
 }
 
+function OvercapacityBanner({ owner }) {
+  if (owner.consumptionRatio < 0.9) return null;
+  const consumptionThreshold =
+    owner.consumptionRatio >= 1.1
+      ? "110%"
+      : owner.consumptionRatio >= 1
+      ? "100%"
+      : "90%";
+
+  return (
+    <Alert
+      severity={owner.consumptionRatio >= 1 ? "danger" : "warning"}
+      mt={-3}
+      mb={3}
+      w="fit-content"
+      mx="auto"
+    >
+      You've hit {consumptionThreshold} of the{" "}
+      {owner.purchases.plan?.name || "Free"} plan limit.{" "}
```

Comment:
purchases should be an array. Also to get the active plan we should not go from purchase. The fallback to free plan must be done server-side.